actionview/lib/action_view/template/types.rb

```diff
@@ -28,30 +28,30 @@ def initialize(symbol)
           @symbol = symbol.to_sym
         end
 
-        delegate :to_s, :to_sym, :to => :symbol
+        def to_s
+          @symbol.to_s
+        end
         alias to_str to_s
 
         def ref
-          to_sym || to_s
+          @symbol
         end
+        alias to_sym ref
 
         def ==(type)
-          return false if type.blank?
-          symbol.to_sym == type.to_sym
+          @symbol == type.to_sym unless type.blank?
         end
       end
 
-      cattr_accessor :type_klass
-
-      def self.delegate_to(klass)
-        self.type_klass = klass
-      end
-
-      delegate_to Type
+      cattr_accessor(:type_klass) { Type }
 
       def self.[](type)
         type_klass[type]
       end
+
+      def symbols
+        type_klass::SET.symbols
```

Comment:
Think you're misunderstanding something. We aren't calling `symbols` on `@symbols`, we're calling it on `TypeMap.new`:

```irb
kaspers-macbook-pro:actionview kasperhansen$ irb
requiirb(main):001:0> require 'set'
=> true
irb(main):002:0> class TypeMap
irb(main):003:1>   attr_reader :symbols
irb(main):004:1> 
irb(main):005:1*   def initialize
irb(main):006:2>     @symbols = Set.new([ :html, :text, :js, :css, :xml, :json ])
irb(main):007:2>   end
irb(main):008:1> end
=> :initialize
irb(main):009:0> SET = TypeMap.new
=> #<TypeMap:0x007fc5f10bb940 @symbols=#<Set: {:html, :text, :js, :css, :xml, :json}>>
irb(main):010:0> SET.symbols
=> #<Set: {:html, :text, :js, :css, :xml, :json}>
```